templates/common/infra/bicep/core/host/container-apps-auth.bicep

```diff
@@ -0,0 +1,42 @@
+metadata description = 'Creates an Azure Container Apps Auth Config using Microsoft Entra as Identity Provider.'
+
+@description('The name of the container apps resource within the current resource group scope')
+param name string
+
+@description('The client ID of the Microsoft Entra application.')
+param clientId string
+
+@description('The name of the Container Apps secret that contains the client secret of the Microsoft Entra application.')
+param clientSecretName string
```

Comment:
Should this be marked with `@secure()` since it is the client secret?

Reply:
It's not the actual secret - it's the name of the secret entry in ACA which contains the actual secret.

See https://github.com/Azure-Samples/chatgpt-quickstart/blob/8dc98f2dca39998f83fde47a5503167529f4b4f0/infra/aca.bicep#L22-L31 for an example of how it's used to configure the secret in an ACA instance.

Reply:
I don't think I understand your first question - it's a separate ARM resource from the app service container so is a separate bicep resource.

As for putting it into container-app-upsert, I did think about it, but wanted to learn a bit more about the patterns of how we'll use it within the Azure Samples first.

Reply:
@jamesc , what I meant was that, instead of doing this:

```bicep
resource container 'Microsoft.App/containerApps@2023-05-02-preview' = {
  name: 'daprstore'
  location: location
  properties: {
      ...
  }
}
resource containerConfig 'Microsoft.App/containerApps/authConfigs@2023-08-01-preview' = {
  name: 'foo'
  parent: container
  properties: {
      ...
  }
}
```

We can do this:

```bicep
resource container 'Microsoft.App/containerApps@2023-05-02-preview' = {
  name: 'daprstore'
  location: location
  properties: {
       ....
  }
  resource foo 'authConfig' = {
    name: 'configurationForAuth'
    properties: {
        ....
    }
  }
}
```

Then, we don't need to bicep modules and a way to connect them by making one to depend on passing the parent to define an existing resource.  Instead, all the functionality for creating appContainer is exposed by one only bicep-module, where you can pass/set parameters to control when to set or not properties, like:

```bicep
resource container 'Microsoft.App/containerApps@2023-05-02-preview' = {
  name: 'daprstore'
  location: location
  properties: {
  }
  resource foo 'authConfigs' = if(authConfigObjectParam) {
    name: 'fosso'
    properties: {
      identityProviders: {

      }
    }
  }
}
```

Reply:
Got it - that seems cleaner. Let me work on that.